src/scrape_up/devpost/devpost.py

```diff
@@ -122,91 +118,38 @@ def get_hackathons(self):
         ```js
         [
             {
-
-                'class_name': 'Software',
-                'name': 'Electron-Cash-SLP',
-                'tagline': 'Electron Cash for SLP Tokens',
-                'slug': 'electron-cash-slp',
-                'url': 'https://devpost.com/software/electron-cash-slp',
-                'members': None,
-                'tags': ['python', 'objective-c', 'c', 'kotlin', 'shell', 'java', 'nsis', 'dockerfile', 'ruby'],
-                'winner': False,
-                'photo': 'https://d112y698adiu2z.cloudfront.net/photos/production/software_thumbnail_photos/002/558/403/datas/small.jpg',
-                'has_video': True,
-                'like_count': 1,
-                'comment_count': 0}
                 'title': 'Ripple CBDC Innovate',
                 'status': '9 days left',
                 'prize': '200,000',
                 'total participants': '1061',
                 'host': 'Ripple',
                 'submission-period': 'May 15 - Aug 18, 2023',
-                'labels': 'Blockchain, Fintech, Social Good, ',
+                'labels': ['Blockchain, Fintech, Social Good'],
                 'hackathon-image': 'https://d112y698adiu2z.cloudfront.net/photos/production/challenge_thumbnails/002/459/630/datas/medium_square.png'
             }
             ...
         ]
         ```
         """
-        url = "https://devpost.com/hackathons"
-
-        try:
-            chrome_options = uc.ChromeOptions()
-            chrome_options.add_argument("--headless")
-            driver = uc.Chrome(options=chrome_options)
-
-            driver.get(url)
-            wait = WebDriverWait(driver, 5)
-        except:
-            return None
-
-        head_tags = wait.until(
-            lambda d: driver.find_elements(By.CLASS_NAME, "hackathon-tile")
-        )
+        url = "https://devpost.com/api/hackathons"
```

Comment:
Hey @iMarble 
We are building an API alternative using scraping to overcome problems like late limiting, session expiration, etc. So it doesn't make any sense to use API under the hood.

Try fetching data using scraping (if possible), else do look for other issue or features you can contribute with.

Reply:
Hey @nikhil25803
The website uses the API under the hood to populate the data. If we need to use scraping - it'll require to wait for that API request to get completed and populate the data in the website (however in this request we are directly using the API endpoint used by the website), which is only possible using the selenium dependency, since it allows to wait for the page to load.

As far as API problems like rate limiting and session expiration are concerned, this will not be an issue since the request is made in the same way as the website.